src/plugins/explore/public/components/query_panel/results_summary/results_summary.tsx

```diff
@@ -0,0 +1,164 @@
+    dataSetState,
+    queryResults,
+    editorMode,
+    queryStatus,
+  } = useSelector((state: RootState) => ({
+    queryState: state.query,
+    lastExecutedPrompt: state.queryEditor.lastExecutedPrompt,
+    dataSetState: state.query.dataset,
+    queryResults: state.results[state.query.query as string]?.hits?.hits,
+    editorMode: state.queryEditor.editorMode,
+    queryStatus: state.queryEditor.overallQueryStatus.status,
+  }));
+
+  const assistantEnabled = core.application.capabilities?.assistant?.enabled;
+  const queryExecutedInPromptMode = useMemo(
+    // Only read the mode when the query result changed
+    () => [EditorMode.DualPrompt, EditorMode.SinglePrompt].includes(editorMode),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [queryResults]
+  );
+
+  // The visibility of panel action buttons: thumbs up/down and copy to clipboard buttons
+  const actionButtonVisible =
+    Boolean(summary) && !loading && queryExecutedInPromptMode && !generateError;
+
+  const fetchSummary = useCallback(
+    async ({ question, query, queryRes }: { question: string; query: string; queryRes: any }) => {
+      if (isEmpty(queryRes) && !isEmpty(summary) && queryStatus !== ResultStatus.READY) {
+        return;
+      }
+
+      setGenerateError(false);
+      setLoading(true);
+      setFeedback(FeedbackStatus.NONE);
+
+      const successMetric = 'generated';
+      try {
+        const actualSampleSize = Math.min(SUMMARY_REQUEST_SAMPLE_SIZE, queryRes?.length);
+        const dataString = JSON.stringify(queryRes?.slice(0, actualSampleSize));
+        const payload = `'${dataString}'`;
+        // TODO: OSD core should not rely on plugin APIs, refactor this once this RFC is
+        // implemented #9859
+        const response = await http.post('/api/assistant/data2summary', {
+          body: JSON.stringify({
+            sample_data: payload,
+            sample_count: actualSampleSize,
+            total_count: queryRes?.length,
+            question,
+            ppl: query,
+          }),
+          query: {
+            dataSourceId: dataSetState?.dataSource?.id,
+          },
+        });
+        setSummary(response);
+        reportCountMetric(successMetric, 1);
+      } catch (error) {
+        reportCountMetric(successMetric, 0);
+        setSummary(
+          i18n.translate('explore.resultsSummary.summary.errorPrompt', {
+            defaultMessage: 'I am unable to respond to this query. Try another question.',
+          })
+        );
+        setGenerateError(true);
+      } finally {
+        setLoading(false);
+      }
+    },
+    [http, reportCountMetric, dataSetState?.dataSource?.id, summary, queryStatus]
+  );
+
+  useEffect(() => {
+    // Close popover and reset summary when a query is re-executed
+    setIsPopoverOpen(false);
+    setSummary('');
+  }, [queryResults, setIsPopoverOpen, setSummary]);
+
+  const autoTriggered = useRef(false);
+  // Trigger auto generating for the first query
```

Comment:
Does it open the popup automatically for the first time?

Reply:
Nope, open the popover is still needed by the user

Reply:
Perhaps we no longer need this `autoTrigger`? because it now always requires user click